controllers/datadogagent/feature/types.go

```diff
@@ -0,0 +1,69 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package feature
+
+import (
+	"github.com/DataDog/datadog-operator/apis/datadoghq/v1alpha1"
+	"github.com/DataDog/datadog-operator/apis/datadoghq/v2alpha1"
+	"github.com/DataDog/datadog-operator/pkg/kubernetes"
+	"github.com/go-logr/logr"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+// Feature Feature interface
+// It returns `true` if the Feature is used, else it return `false`.
+type Feature interface {
+	// Configure use to configure the internal of a Feature
+	// It should return `true` if the feature is enabled, else `false`.
+	Configure(dda *v2alpha1.DatadogAgent) bool
```

Comment:
Is this function both altering the DatadogAgent spec and checking if the feature is enabled?

Reply:
From the ksmcore example it seems only the latter, in which case maybe we can call it `IsConfigured` or `IsEnabled`

Reply:
yes the idea it to extra the information needed from the `DatadogAgent`. like this we don't need to access again the resource after when processing the feature